src/api/event.ts

```diff
@@ -16,13 +17,66 @@ export class Event {
   }
 
   /**
-   * Get events by creation number and the address
+   * Get events by creation number and an account address
    *
    * @param args.address - The account address
    * @param args.creationNumber - The event creation number
-   * @returns Promise<GetEventsByCreationNumberResponse>
+   *
+   * @returns Promise<GetEventsResponse>
+   */
+  async getAccountEventsByCreationNumber(args: {
+    address: string;
+    creationNumber: AnyNumber;
+    options?: {
+      pagination?: IndexerPaginationArgs;
+      orderBy?: OrderBy<GetEventsResponse[0]>;
+    };
+  }): Promise<GetEventsResponse> {
+    return getAccountEventsByCreationNumber({ aptosConfig: this.config, ...args });
+  }
+
+  /**
+   * Get events by event type and an account address
+   *
+   * @param args.address - The account address
+   * @param args.eventType - The event type
+   *
+   * @returns Promise<GetEventsResponse>
+   */
+  async getAccountEventsByEventType(args: {
+    address: string;
+    eventType: MoveResourceType;
+    options?: {
+      pagination?: IndexerPaginationArgs;
+      orderBy?: OrderBy<GetEventsResponse[0]>;
+    };
+  }): Promise<GetEventsResponse> {
+    return getAccountEventsByEventType({ aptosConfig: this.config, ...args });
+  }
+
+  /**
+   * Get all events
+   *
+   * An optional `where` can be passed in to filter out the response.
+   *
+   * @example
+   * ```
+   * { where:
+   *  {
+   *   transaction_version: { _eq: 123456 },
+   *  }
+   * }
+   * ```
+   *
+   * @returns GetEventsQuery response type
    */
-  async getEventsByCreationNumber(args: { address: string; creationNumber: AnyNumber }): Promise<GetEventsResponse> {
-    return getEventsByCreationNumber({ aptosConfig: this.config, ...args });
+  async getEvents(args?: {
+    options?: {
+      where?: EventsBoolExp;
+      pagination?: IndexerPaginationArgs;
+      orderBy?: OrderBy<GetEventsResponse[0]>;
```

Comment:
I'm curious what the [0] is in this context since I'm learning these random things in typescript

Reply:
So GetEventsResponse is technically array - e.g. {prop1: number, prop2: string ....}[]

We want the type {prop1: number, prop2: string ....} .  The [0] allows us to reference the underlying type.

I asked the same thing to @0xmaayan recently lol.

Reply:
yes, this is a design choice. 
The response is an array of items. We want to order by every item in the array and therefore we use the first element keys to know what we can order by.